rust/sdk-processor/src/steps/parquet_default_processor/gcs_handler.rs

```diff
@@ -0,0 +1,185 @@
+use crate::parquet_processors::{ParquetTypeEnum, ParquetTypeStructs};
+use anyhow::Context;
+use aptos_indexer_processor_sdk::utils::errors::ProcessorError;
+use async_trait::async_trait;
+use google_cloud_storage::client::Client as GCSClient;
+use parquet::{
+    file::{properties::WriterProperties, writer::SerializedFileWriter},
+    record::RecordWriter,
+    schema::types::Type,
+};
+use processor::bq_analytics::{
+    gcs_handler::upload_parquet_to_gcs,
+    generic_parquet_processor::{GetTimeStamp, HasParquetSchema, HasVersion},
+};
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+use tracing::{debug, error};
+
+pub struct GCSUploader {
```

Comment:
Should this be called DefaultGCSUploader? Let's rename if other parquet processors are expected to upload to GCS. This follows the convention of the postgres processors in this dir

Reply:
agreed, this will be shared by other parquet processors.

Reply:
just to clarify, I was hoping that we can extend this GCS uploader to support other parquet types as well instead of creating GCS per processor

Reply:
oh. interesting. what do you mean by extending it?